geode-core/src/test/java/org/apache/geode/internal/cache/partitioned/ClearPRMessageTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.cache.partitioned;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.apache.geode.distributed.DistributedLockService;
+import org.apache.geode.internal.cache.BucketRegion;
+import org.apache.geode.internal.cache.ForceReattemptException;
+import org.apache.geode.internal.cache.PartitionedRegion;
+import org.apache.geode.internal.cache.PartitionedRegionDataStore;
+import org.apache.geode.internal.cache.PartitionedRegionException;
+
+public class ClearPRMessageTest {
```

Comment:
It might be worth including tests to confirm that `initMessage()` correctly sets the `processorId` and doesn't call `enableSevereAlertProcessing()` on the `ReplyProcessor` if the processor is null, a test to confirm that `send()` throws an exception if `putOutgoing(this)` returns a non-null set, a test to confirm that `operateOnPartitionedRegion()` calls `sendReply()` with the correct arguments if a `ForceReattemptException` is thrown during `doLocalClear()`, a test to confirm that we call `endPartitionMessagesProcessing()` in `sendReply()` when appropriate and a test to confirm that we only call `replyProcessor.process()` in `ClearReplyMessage.process()` when the replyProcessor is not null.

Reply:
I added all these tests - a few of them were just making sure that methods weren't called on fields that were null, and these tests felt a little unnecessary (they pretty much just involved calling the method and relying on an NPE to fail) but I've added them anyway just to be safe. Let me know if you think these are unnecessary.